tests/unit/ResolutionDates.spec.ts

```diff
@@ -205,6 +206,72 @@ describe('Resolution Dates component - edit mode', () => {
 
     wrapper.destroy()
   })
+
+  it('displays the error message when invalid', async () => {
+    const wrapper = wrapperFactory()
+
+    // verify there is just 1 row
+    const rows = wrapper.findAll('.row')
+    expect(rows.length).toBe(1)
+
+    // verify first row has 3 columns
+    const cols = rows.at(0).findAll('.col')
+    expect(cols.length).toBe(3)
+
+    // verify column text
+    expect(cols.at(1).text()).toContain('You must add a resolution or court order date')
+
+    wrapper.destroy()
+  })
+
+  it('hides the error message when valid', async () => {
+    const wrapper = wrapperFactory()
+    const vm = wrapper.vm
+
+    // verify there is just 1 row
+    let rows = wrapper.findAll('.row')
+    expect(rows.length).toBe(1)
+
+    // verify first row has 3 columns
+    const cols = rows.at(0).findAll('.col')
+    expect(cols.length).toBe(3)
+
+    // verify column text contains error initially
+    expect(cols.at(1).text()).toContain('You must add a resolution or court order date')
+
+    // click the Add button
+    await vm.$el.querySelector('.add-btn').click()
+
+    // verify there are now 2 rows
+    rows = wrapper.findAll('.row')
+    expect(rows.length).toBe(2)
+
+    // verify date picker form
+    expect(wrapper.find('.date-picker-form')).toBeDefined()
+
+    // inject a date
+    vm.onDateEmitted('2021-03-17')
+    store.state.stateModel.shareStructureStep.resolutionDates = ['2021-03-17']
+    await Vue.nextTick()
```

Comment:
Great test. I bet you had fun with this one! :)

Reply:
haha always do. Tests not only test the code but our resolve.